### PR TITLE
Add script that can do retry when git clone-ing

### PR DESCRIPTION
**Pull request by MattGal**
After wasting a lot of time trying to do this with inline powershell (just doesn't fit into what's allowed) I've made a .cmd version of this for running git clone with retry.  Additionally, I've made the clone default to --depth 1 to improve performance.  Retry count, sleep time, and depth can be made into parameters later if desired.
(There's little / no reason to have a non-Windows version of this script as there is a built-in git-retry command for *nix)

@chcosta

### CODE REVIEW

**Comment by mmitche** [comment]
@MattGal Clone depth = 1 breaks some repos

**Comment by MattGal** [comment]
@mmitche is there a safe depth we could pick here?  in experimenting this seems fine with CoreFX.  I added this to the command as a perf improvement on Wes' suggestion.

**Comment by mmitche** [comment]
I don't think there was.  Let me find out who was using it.  They use depth to figure out a build number.

**Comment by mmitche** [comment]
@MattGal Core setup.  Forwarded mail thread to you.

**Comment by MattGal** [comment]
@chcosta OK to merge?  I'd like to try this as a workaround for the clone timeouts we've been seeing in CoreFX.

**Comment by karajas** [comment]
Do you want to add verbose logging just in case? 
```
git clone repo --depth 1 -v
```

**Comment by MattGal** [comment]
@karajas I haven't actually seen a case where this would have helped.  The default level of verbosity has been sufficient thus far and is what's used in the build steps I'd like to replace this with.

**Comment by chcosta** [comment]
How are you intending to launch this script?  ie, you don't have access to buildtools until after you've cloned the repo.  Unless you have some other plan to download the file directly and execute it for the cloning?

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/scripts/vstsagent/CloneWithRetry.cmd [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+echo retrying up to 10x if there is a failure.
+goto :eof
+
+:main
+set REPO_URL=%1
+set FOLDER_NAME=%2
+set /a RETRY_COUNT=10
+set /a SLEEP_TIME=10
+
+IF NOT DEFINED GIT SET "GIT=%PROGRAMFILES(X86)%\Git\cmd\git.exe"
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/scripts/vstsagent/CloneWithRetry.cmd]: Is GIT the env we use for finding it? Should we consider the version on the path?

**Comment by weshaggard** [comment]
>@MattGal Core setup. Forwarded mail thread to you.

yes core-setup does use commit count for their versioning currently, so does CLI but that will likely be changing at least for core-setup.

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/scripts/vstsagent/CloneWithRetry.cmd (hunk 1 above)
Comment (MattGal) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/scripts/vstsagent/CloneWithRetry.cmd]: GIT was the environment variable that the build step I was hoping to use this in was already using, the goal was to change nothing about this, but yeah could just change it to IF NOT DEFINED GIT SET GIT=git.exe and use the version on the path.   However, I'm realizing that if this is in buildtools it won't be available before the repo is cloned, brainstorming offline with chcosta now.

**Comment by MattGal** [comment]
Closing for now, will continue to think about how to reasonable bake some retry logic into cloning.

**Pull request closed without merging**